src/AspNetCoreModuleV2/CommonLib/FileOutputManager.cpp

```diff
@@ -0,0 +1,194 @@
+            systemTime.wMinute,
+            systemTime.wSecond,
+            GetCurrentProcessId(),
+            fsPath.filename().stem().c_str()));
+
+    saAttr.nLength = sizeof(SECURITY_ATTRIBUTES);
+    saAttr.bInheritHandle = TRUE;
+    saAttr.lpSecurityDescriptor = NULL;
+
+    // Create the file with both READ and WRITE.
+    m_hLogFileHandle = CreateFileW(m_struLogFilePath.QueryStr(),
+        FILE_READ_DATA | FILE_WRITE_DATA,
+        FILE_SHARE_READ,
+        &saAttr,
+        CREATE_ALWAYS,
+        FILE_ATTRIBUTE_NORMAL,
+        nullptr);
+
+    RETURN_LAST_ERROR_IF(m_hLogFileHandle == INVALID_HANDLE_VALUE);
+
+    stdoutWrapper = std::make_unique<StdWrapper>(stdout, STD_OUTPUT_HANDLE, m_hLogFileHandle, m_fEnableNativeRedirection);
+    stderrWrapper = std::make_unique<StdWrapper>(stderr, STD_ERROR_HANDLE, m_hLogFileHandle, m_fEnableNativeRedirection);
+
+    stdoutWrapper->StartRedirection();
+    stderrWrapper->StartRedirection();
+
+    return S_OK;
+}
+
+
+HRESULT
+FileOutputManager::Stop()
+{
+    STRA     straStdOutput;
+    CHAR            pzFileContents[MAX_FILE_READ_SIZE] = { 0 };
+    DWORD           dwNumBytesRead;
+    LARGE_INTEGER   li = { 0 };
+    DWORD           dwFilePointer = 0;
+
+    if (m_disposed)
+    {
+        return S_OK;
+    }
+
+    SRWExclusiveLock lock(m_srwLock);
+
+    if (m_disposed)
+    {
+        return S_OK;
+    }
+
+    m_disposed = true;
+
+    HANDLE   handle = NULL;
+    WIN32_FIND_DATA fileData;
+
+    if (m_hLogFileHandle != INVALID_HANDLE_VALUE)
+    {
+        FlushFileBuffers(m_hLogFileHandle);
+    }
+
+    if (stdoutWrapper != nullptr)
+    {
+        stdoutWrapper->StopRedirection();
+    }
+
+    if (stderrWrapper != nullptr)
+    {
+        stderrWrapper->StopRedirection();
+    }
+
+    // delete empty log file
+    handle = FindFirstFile(m_struLogFilePath.QueryStr(), &fileData);
+    if (handle != INVALID_HANDLE_VALUE &&
+        handle != NULL &&
+        fileData.nFileSizeHigh == 0 &&
+        fileData.nFileSizeLow == 0) // skip check of nFileSizeHigh
+    {
+        FindClose(handle);
+        LOG_LAST_ERROR_IF(!DeleteFile(m_struLogFilePath.QueryStr()));
+    }
+
+    // Read the first 30Kb from the file and store it in a buffer.
+    // By doing this, we can close the handle to the file and be done with it.
+    if (m_hLogFileHandle != INVALID_HANDLE_VALUE)
+    {
+        if (GetFileSizeEx(m_hLogFileHandle, &li) && li.LowPart > 0 && li.HighPart == 0)
+        {
+            dwFilePointer = SetFilePointer(m_hLogFileHandle, 0, NULL, FILE_BEGIN);
+
+            if (dwFilePointer != INVALID_SET_FILE_POINTER)
+            {
+                // Read file fails.
+                if (ReadFile(m_hLogFileHandle, pzFileContents, MAX_FILE_READ_SIZE, &dwNumBytesRead, NULL))
+                {
+                    m_straFileContent.Copy(pzFileContents, dwNumBytesRead);
+                }
+            }
+        }
+    }
+
+    if (GetStdOutContent(&straStdOutput))
```

Comment:
Why do you write to m_straFileContent in this method but call into GetStdOutContent to get it's value back?